### PR TITLE
se agrego todo lo correspondiente a la autenticación 2fa

### PR DESCRIPTION
**Pull request by StivenLH24**

### CODE REVIEW

**Review by Bycristian000 — APPROVED** [review]

**Review by jonvas06 — APPROVED** [review]

File: src/services/security-user.service.ts [hunk 1]
```diff
@@ -1,33 +1,53 @@
...
 const generator = require('generate-password');
 const MD5 = require('crypto-js/md5');
 
 @injectable({scope: BindingScope.TRANSIENT})
 export class SecurityUserService {
-  constructor(/* Add @inject to inject parameters */) {}
-  
+  constructor(
+    @repository(UserRepository)
+    public repositoryuser: UserRepository,
```
Comment (Eider22) [inline src/services/security-user.service.ts:12]: use cammellCase   → repositoryUser

File: src/controllers/user.controller.ts [hunk 2]
```diff
@@ -156,4 +152,44 @@ export class UserController {
...
+           'aplication/json':{
+             schema: getModelSchemaRef(Credentials)
+           }
+         }
+       }
+     )
+     credentials : Credentials
+   ): Promise<object> {
+    let user = this.serviceSecurity.identifyUser(credentials);
+    if (user) {
```
Comment (Eider22) [inline src/controllers/user.controller.ts:179]: Change structure if-else.
First evaluates escape condition and "else"  isn´t necesary.

Something like:
if(!user){
    return new HttpErrors[401]("incorrect credentials.");
}


let code2fa = this.serviceSecurity.createTextRandom(5);
      let login: Login = new Login();
      login.userId = user._id!;
      login.code2fa = code2fa;
      login.codeState2fa = false;
      login.token = "";
      login.tokenState = false;
      this.repositoryLogin.create(login);
      // notify the user via mail or sms
      return user;

File: src/controllers/user.controller.ts [hunk 3]
```diff
@@ -156,4 +152,44 @@ export class UserController {
...
+      login.userId = user._id!;
+      login.code2fa = code2fa;
+      login.codeState2fa = false;
+      login.token = "";
+      login.tokenState = false;
+      this.repositoryLogin.create(login);
+      // notify the user via mail or sms
+      return user;
+    }
+    return new HttpErrors[401]("incorrect credentials.");
```
Comment (Eider22) [inline src/controllers/user.controller.ts:191]: change "return" by  "throw"

File: src/controllers/user.controller.ts [hunk 4]
```diff
@@ -156,4 +152,44 @@ export class UserController {
...
+         content:{
+           'aplication/json':{
+             schema: getModelSchemaRef(Credentials)
+           }
+         }
+       }
+     )
+     credentials : Credentials
+   ): Promise<object> {
+    let user = this.serviceSecurity.identifyUser(credentials);
```
Comment (Eider22) [inline src/controllers/user.controller.ts:178]: Use try catch

**Review by Eider22 — CHANGES_REQUESTED** [review]

File: src/controllers/user.controller.ts [hunk 5]
```diff
@@ -156,4 +152,44 @@ export class UserController {
   async deleteById(@param.path.string('id') id: string): Promise<void> {
     await this.userRepository.deleteById(id);
   }
+
+   /**
+ * metodos personalizados para la Api
+ *
+ */
+
+   @post('/identify-users')
```
Comment (Bycristian000) [inline src/controllers/user.controller.ts:161]: seria identify-user , segun video que esta como "identificar usuario"

File: src/controllers/user.controller.ts [hunk 6]
```diff
@@ -1,33 +1,29 @@
-import {service} from '@loopback/core';
+import {service} from '@loopback/cnpmore';
```
Comment (Eider22) [inline src/controllers/user.controller.ts:1]: Why this dependency?

File: src/controllers/user.controller.ts [hunk 7]
```diff
@@ -1,33 +1,29 @@
-import {service} from '@loopback/core';
```
Comment (Eider22) [inline src/controllers/user.controller.ts:1]: Why delete this?

**Review by Eider22 — APPROVED** [review]

**Pull request merged**